src/lua/LuaInterfaceGame.h

```diff
@@ -43,6 +44,10 @@ class LuaInterfaceGame : public LuaInterfaceBase
     void EventStart(bool isFirstStart);
     void EventGameFrame(unsigned number);
     void EventResourceFound(unsigned char player, const MapPoint pt, unsigned char type, unsigned char quantity);
+    // Called if player wants to cancel a pact
+    bool EventCancelPact(PactType pt, unsigned char canceledByPlayerId, unsigned char targetPlayerId);
+    // Called if pact is suggested
+    void EventSuggestPact(unsigned int startFrame, PactType pt, unsigned char suggestedByPlayerId, unsigned char targetPlayerId, unsigned int duration);
```

Comment:
Implementation is missing (was removed). Check the iwPostWindow for correct handling. And don't pass the GF to lua. It is only required by the response handling.

Reply:
Yeah is had a not working implementation for that, and forgot to remove it from header file.
iwPostWindow calls GAMECLIENT as GameCommandFactory. There is the problem, that if i import GAMECLIENT as singleton instance (from header file), the GameCommandFactory builds correct instances for the host and not for the AI. 
So this game commands are not valid for the AI (tries to suggest a pact between player 0 and player 0).

is there any way to load a GameCommandFactory for another player?

About passing the GF:
While accepting the pact, the script needs to know the PactID (which is also the GF when the pact was suggested, if im right). So where should get script get knowledge of this id instead of this event handle? PactID would be a better naming. On the otherhand i could generate a "custom" id and remap this to the GF?

Reply:
Damn, you are right. One would need the actual AI player for that (currently in GAMESERVER)

Not really: You call the lua function and that returns true or false. Then you call the accept with the GF you still have.